models/idea/index.js

```diff
@@ -308,6 +308,52 @@ class Idea extends Model {
     const cursor = await this.db.query(query, params);
     return await cursor.all();
   }
+
+
+  /**
+   * Read trending ideas
+   * @param {integer} offset - pagination offset
+   * @param {integer} limit - pagination limit
+   * @returns {Promise<Idea[]>} - list of found ideas
+   */
+  static async findTrending({ offset, limit }) {
+    const now = + new Date();
+    const oneWeek = 604800000; // 1000 * 60 * 60 * 24 * 7
+    const threeWeeks = 1814400000; // 1000 * 60 * 60 * 24 * 21
+    const threeMonths = 7776000000; // 1000 * 60 * 60 * 24 * 90
+    const weekAgo = now - oneWeek;
+    const threeWeeksAgo = now - threeWeeks;
+    const threeMonthsAgo = now - threeMonths;
+
+    // for each idea we are counting 'rate'
+    // rate is the sum of votes/day in the last three months
+    // votes/day from last week are taken with wage 3
+    // votes/day from two weeks before last week are taken with wage 2
+    // votes/day from the rest of days are taken with wage 1
+    const query = `
+      FOR idea IN ideas
+        FOR vote IN votes
+        FILTER idea._id == vote._to
+        // group by idea id
+        COLLECT id = idea
+        // get sum of each idea's votes values from last week, last three weeks and last three months
+        AGGREGATE rateWeek = SUM((vote.value * TO_NUMBER( @weekAgo <= vote.created))/7),
+                rateThreeWeeks  = SUM((vote.value * TO_NUMBER( @threeWeeksAgo <= vote.created <= @weekAgo))/21),
```

Comment:
Is the 7, 21, 90 here meant to have `voteSum` per day during the period in question? Makes complete sense. :+1:  Though perhaps 7, 14, 69 should be used instead, because the period has a bottom time limit, too.